### PR TITLE
Add functions

### PR DESCRIPTION
**Pull request by jayspeidell**
## Description
Objective Function Updates: Verified Goldstein and Booth functions working and removed TODO tags. Fixed Bulkin No 6 and removed TODO tag. Implemented Cross-in-Tray, Easom, Eggholder, Himmelblau's, Holder Table, and Three Hump Camel functions. File is getting long so I sorted the functions and added a list of functions and their corresponding function name to the docstring.

Bug fix: Added 'z-axis' to formatters/Designer attribute 'labels' to prevent index out of range error when plotting in 3D.

New feature: It's hard to visualize some functions with a continuous colormap, so I added a colormaps property to the Designer class in formatters.py to allow the user to choose whichever one they want. It uses the same style as other properties of the class, and the module imports matplotlib.cm to validate both types of colormaps in matplotlib. I edited plotters/plot_countour to initialize a Designer object with a default colormap if none is provided. I've tested with every category of colormap as well as custom colormaps and verified that it is working as intended. I made the edit because I personally find it easier to understand the contours when they use a discreet colormap.

Default parameter change: Mesher.delta from 0.001 to 0.1. This makes it safe to run many of the functions on an average computer. With 0.001, a computer with 8GB of RAM will crash above a domain size of [8, 8]. Increasing by two orders of magnitude allows most provided functions to run without crashing due to memory errors and doesn't visibly impact graph quality. 

## Related Issue

Add more benchmark functions #158

Fixing a bug: Designer only had two axis labels by default, so I added "z-axis". 

Added a feature: Designer accepts a colormap from the user. 

Fixing memory error: The Mesher.delta parameter of 0.001 has a high chance of causing crashes to memory errors. 0.1 is safer for most of the functions in single_obj.py, but some will still crash. 

## Motivation and Context

Adding a colormap attribute to Designer made easier to verify the functions I implemented because it enabled me to use discreet colormaps. 

I fixed the range out of error bug because it caused a crash. 

I changed the default parameter of Mesher.delta because it uses a massive amount of memory. 

## How Has This Been Tested?

I created a notebook with all the added functions and tested each function as I implemented it. All functions function in the full domain and visually match examples found online, and the minimums are what they should be. 

Then I used flask8 and black. I restricted flask8 and black testing to the files that I edited in each commit. 

I also ran py.test. It got a lot of warnings, but I made sure that my edits didn't add any new ones on top of those that occur in a fresh clone of the development branch. 

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by whzup** [comment]
Very nice @jayspeidell! Gonna review this one in the evening. Very excited to do so!

File: pyswarms/utils/functions/single_obj.py [hunk 1]
```diff
@@ -133,20 +229,45 @@ def rosenbrock_func(x):
...
+def easom_func(x):
+    """Easom objective function.
 
-    Only takes two dimensions and has a global minimum at
-    :code:`f([3,0.5])` Its domain is bounded between :code:`[-4.5, 4.5]`
+    Only takes two dimensions and has a global minimum of
+    `-1` at :code:`f([pi, pi])`.
+    Its coordinates are bounded within :code:`[-100,100]`.
+
+    Best visualized in thedomain of :code:`[-5, 5]` and a range of :code:`[-1, 0.2]`.
```
Comment (whzup) [inline pyswarms/utils/functions/single_obj.py]: the domain*

**Comment by ljvmiranda921** [comment]
Hi @jayspeidell 

Thank you for your contribution and welcome to PySwarms! :+1: I noticed that you are working with both the single-objective functions and the plotters library in the same Pull Request. I'd recommend to:
1. Keep *only* the single-objective-related changes in this PR, and 
2. Open up an Issue + new PR for the changes in the `plotters` library.

We want to keep all changes as atomic as possible, so we can have a concise history of the project :+1: . I know it may be a bit of a hassle for you but if you need any help, we're always available!

So this is what should happen:
```
development (branch) -------
                           \---------- jayspeidell:add-functions
                           \---------- jayspeidell:fix-plotters-module          
```
Both branches will be merged into the development branch after review :+1: 

## In your local machine
In Git, what you can do is:
1. Make sure you have a local copy of all your changes as backup
2.  [`revert`](https://www.atlassian.com/git/tutorials/undoing-changes/git-revert) all commits related to the plotters module, **or** do [`git rebase`](https://www.atlassian.com/git/tutorials/rewriting-history#git-rebase-i) and delete them.
3. Push changes to `jayspeidell:add-functions`, making sure that only changes in the single-objective functions (or specifically, `single_obj.py`) were done.
4. Open up a new Issue, describing the fix and changes needed for the `plotters` module. (I know this is a hassle, but we really need a well-documented history of the project :+1: )
5. Make a new branch in your own fork,  `jayspeidell:fix-plotters-module`
6. Copy or do all changes only related to the `plotters` module
7. Make a new Pull Request! :100: 

And that's it! 

Thank you for your help and we really appreciate your contribution to PySwarms! Take your time and feel free to ask us if you have any questions!

**Comment by jayspeidell** [comment]
Thanks for the feedback! I'll create the "fix plotters module" issue now and then split the two contributions into separate branches. 

Before I do another pull request, how do I make sure that the pull request references a specific issue?

**Comment by whzup** [comment]
Hey @jayspeidell! You can just do it as you did here with a link to the issue (a hashtag followed by the number the linking is automatic).
BTW, I saw that tests are missing for your functions. Please add tests for your new functions to these files [here](https://github.com/ljvmiranda921/pyswarms/tree/master/tests/utils/functions). You can copy a test function and paste it, then you can customize it to fit one of you new functions, repeat until done 😋. You can run the tests by navigating into the directory I mentioned in your CLI and then using `pytest`.

**Comment by ljvmiranda921** [comment]
Hi @jayspeidell 

> I'll create the "fix plotters module" issue now and then split the two contributions into separate branches.

Awesome! We do this so that we have a well-documented history of all problems, bugs, etc. in the project

> Before I do another pull request, how do I make sure that the pull request references a specific issue?

Make sure that the commits for one PR is only concerned with a specific issue (i.e, one for #158 and another for the one you'll create). 

I believe that for the single-objective functions PR, the only file that should change will be the `single_obj.py` (because you added new functions) and `test_single_obj.py` (because of new tests).
For another PR, I assume that only the files related to the `plotters` module should change. 

You can check your commits at the "Commits" tab near the PR title. 

So again: 
 - one branch is only for single-objective functions, and 
 - another branch for fixing the plotters module. 

Both of them should **branch from** the `development` branch :+1: 
This may be a loose example, but this is how i think the two branches should be made:

```shell
$ # Currently in development branch
$ git checkout -b add-functions # Creates add-functions branch and jumps there
$ # Add all changes related to add-functions
$ git checkout development # Goes back to the development branch
$ git checkout -b fix-plotters-module # Creates fix-plotters-module branch and jumps there
$ # Add all changes/fixes related to fix-plotters-module
```

Then you can push the changes in your two branches and do two PRs :+1:

**Comment by jayspeidell** [comment]
Got it! Sorry, I made the pull request before seeing your message about the functions.

**Pull request closed without merging**